static/app/views/settings/projectTags/index.tsx

```diff
@@ -7,140 +7,146 @@ import {Button} from 'sentry/components/button';
 import Confirm from 'sentry/components/confirm';
 import EmptyMessage from 'sentry/components/emptyMessage';
 import ExternalLink from 'sentry/components/links/externalLink';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
 import Panel from 'sentry/components/panels/panel';
 import PanelBody from 'sentry/components/panels/panelBody';
 import PanelHeader from 'sentry/components/panels/panelHeader';
 import PanelItem from 'sentry/components/panels/panelItem';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {IconDelete} from 'sentry/icons';
 import {t, tct} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
-import {Organization, Project, TagWithTopValues} from 'sentry/types';
+import {TagWithTopValues} from 'sentry/types';
+import {
+  setApiQueryData,
+  useApiQuery,
+  useMutation,
+  useQueryClient,
+} from 'sentry/utils/queryClient';
+import RequestError from 'sentry/utils/requestError/requestError';
 import routeTitleGen from 'sentry/utils/routeTitle';
-import DeprecatedAsyncView from 'sentry/views/deprecatedAsyncView';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
 import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
 import TextBlock from 'sentry/views/settings/components/text/textBlock';
 import PermissionAlert from 'sentry/views/settings/project/permissionAlert';
 
-type Props = RouteComponentProps<{projectId: string}, {}> & {
-  organization: Organization;
-  project: Project;
-} & DeprecatedAsyncView['props'];
-
-type State = {
-  tags: Array<TagWithTopValues>;
-} & DeprecatedAsyncView['state'];
-
-class ProjectTags extends DeprecatedAsyncView<Props, State> {
-  getDefaultState(): State {
-    return {
-      ...super.getDefaultState(),
-      tags: [],
-    };
-  }
+type Props = RouteComponentProps<{projectId: string}, {}>;
+
+type DeleteTagResponse = unknown;
+type DeleteTagVariables = {key: TagWithTopValues['key']};
+
+function ProjectTags(props: Props) {
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const {projectId} = props.params;
+
+  const project = projects.find(p => p.id === projectId);
+
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  const {
+    data: tags,
+    isLoading,
+    isError,
+  } = useApiQuery<TagWithTopValues[]>(
+    [`/projects/${organization.slug}/${projectId}/tags/`],
+    {staleTime: 0}
+  );
+
+  const {mutate} = useMutation<DeleteTagResponse, RequestError, DeleteTagVariables>({
+    mutationFn: ({key}: DeleteTagVariables) =>
+      api.requestPromise(`/projects/${organization.slug}/${projectId}/tags/${key}/`, {
+        method: 'DELETE',
+      }),
+    onSuccess: (_, {key}) => {
```

Comment:
One more thing to add here is an `onError` that adds an error toast message. I think it's `addErrorMessage('message')`